### DISCUSSION TITLE
求助：mail作业中在本地测试通过，上传到git上测试时在task7 failed

### DISCUSSION THREAD

**Discussion by HrZhang18** [physics-data/faq#18] [Q&A] [answered]
我本地使用的系统是MacOS, 在本地测试和在git上测试唯一不同的是把所有的wc, cut命令换成了gwc, gcut，还有把sed -i换成了gsed 
。在本地使用grade.sh顺利通过了测试，但在git上auto-grade时在task7failed了。
[logs_16.zip](https://github.com/physics-data/faq/files/5173071/logs_16.zip)
这是我在git上测试时的log。

**Comment by HrZhang18** [comment]
另外，在本地我还将grade.sh中的WC、AWK变量设置为了gwc和gawk

**Comment by ZenithalHourlyRate** [comment]
在Makefile中有个目标的写法出现了问题。
你的某个目标没有 dependency，所以在每次执行时，该目标都会被执行一次；我们要求的是，该目标只在被需要的时候才被执行。
「这个脚本需要且必须在其他脚本运行前执行」这句话的意思是，该目标并非一定要被执行。

**Comment by ZenithalHourlyRate** [comment, marked as answer]
抱歉我看错了，又星际玩家了一次。
你的dependency里面可能出现了一些问题。
你可以注意到，imp在最后执行的target中会被改动，而最后的target依赖这个imp，那么你的make会重复执行最后这个target
可以改动依赖的书写方式
